actions/app.go

```diff
@@ -29,31 +37,53 @@ func App() *buffalo.App {
 			},
 			SessionName: "_rdsapi_session",
 		})
-		// Automatically redirect to SSL
-		app.Use(forceSSL())
-
-		// Set the request content type to JSON
-		app.Use(middleware.SetContentType("application/json"))
 
 		if ENV == "development" {
-			app.Use(middleware.ParameterLogger)
+			app.Use(paramlogger.ParameterLogger)
+		}
+
+		// load json config
+		AppConfig, _ := common.LoadConfig("config/config.json")
+
+		// create a shared RDS session for each account
+		for account, c := range AppConfig.Accounts {
+			RDS[account] = rds.NewSession(c) //.Akid, c.Secret, c.Region)
 		}
 
-		app.GET("/", HomeHandler)
+		app.GET("/v1/rds/ping", PingPong)
 
+		rdsV1API := app.Group("/v1/rds/{account}")
+		rdsV1API.Use(sharedTokenAuth(AppConfig.Token), validateAccount)
+		rdsV1API.POST("/", DatabasesPost)
+		rdsV1API.GET("/", DatabasesGet)
+		rdsV1API.GET("/{db}", DatabasesGet)
+		rdsV1API.DELETE("/{db}", DatabasesDelete)
 	}
 
 	return app
 }
 
-// forceSSL will return a middleware that will redirect an incoming request
-// if it is not HTTPS. "http://example.com" => "https://example.com".
-// This middleware does **not** enable SSL. for your application. To do that
-// we recommend using a proxy: https://gobuffalo.io/en/docs/proxy
-// for more information: https://github.com/unrolled/secure/
-func forceSSL() buffalo.MiddlewareFunc {
-	return ssl.ForceSSL(secure.Options{
-		SSLRedirect:     ENV == "production",
-		SSLProxyHeaders: map[string]string{"X-Forwarded-Proto": "https"},
-	})
+// sharedTokenAuth middleware validates the auth token
+func sharedTokenAuth(token string) buffalo.MiddlewareFunc {
+	return func(next buffalo.Handler) buffalo.Handler {
+		return func(c buffalo.Context) error {
+			headers, ok := c.Request().Header["X-Auth-Token"]
+			if !ok || len(headers) == 0 || headers[0] != token {
+				log.Println("Missing or bad token header for request", c.Request().URL)
+				return c.Error(403, errors.New("Forbidden"))
+			}
+			return next(c)
+		}
+	}
+}
+
+// validateAccount middleware validates the account
+func validateAccount(next buffalo.Handler) buffalo.Handler {
+	return func(c buffalo.Context) error {
+		if _, ok := RDS[c.Param("account")]; !ok {
+			log.Printf("Account not found: %s", c.Param("account"))
+			return c.Error(400, errors.New("Bad request: unknown account "+c.Param("account")))
+		}
+		return next(c)
+	}
```

Comment:
I question this middleware a little only because I like to see the map validation inline... maybe we should use both.

Reply:
I figured since that exact check is required for all handlers it makes sense to separate it here.